src/main/scala/cognite/spark/v1/LabelsRelation.scala

```diff
@@ -0,0 +1,70 @@
+package cognite.spark.v1
+
+import java.time.Instant
+
+import cats.effect.IO
+import cognite.spark.v1.SparkSchemaHelper._
+import com.cognite.sdk.scala.v1.{GenericClient, Label, LabelCreate, LabelsFilter}
+import org.apache.spark.sql.sources.{Filter, InsertableRelation}
+import org.apache.spark.sql.types._
+import org.apache.spark.sql.{Row, SQLContext}
+import cats.implicits._
+
+class LabelsRelation(config: RelationConfig)(val sqlContext: SQLContext)
+    extends SdkV1Relation[Label, String](config, "labels")
+    with InsertableRelation
+    with WritableRelation {
+
+  override def schema: StructType = structType[Label]
+
+  override def toRow(a: Label): Row = asRow(a)
+
+  override def uniqueId(a: Label): String = a.externalId
+
+  override def getStreams(filters: Array[Filter])(
+      client: GenericClient[IO],
+      limit: Option[Int],
+      numPartitions: Int): Seq[fs2.Stream[IO, Label]] =
+    Seq(client.labels.filter(LabelsFilter()))
+
+  override def insert(rows: Seq[Row]): IO[Unit] = {
+    val labels = rows.map(fromRow[LabelCreate](_))
+    client.labels
+      .create(labels)
+      .flatTap(_ => incMetrics(itemsCreated, labels.length)) *> IO.unit
+  }
+
+  override def delete(rows: Seq[Row]): IO[Unit] = {
+    val labelIds = rows.map(fromRow[LabelDeleteSchema](_)).map(_.externalId)
+    client.labels
+      .deleteByExternalIds(labelIds)
+      .flatTap(_ => incMetrics(itemsDeleted, labelIds.length))
+  }
+
+  override def upsert(rows: Seq[Row]): IO[Unit] = ???
+
+  override def update(rows: Seq[Row]): IO[Unit] = ???
```

Comment:
Not quite sure what to do with these. The labels API doesn't support updates, but having the `WritableRelation` made deletes and inserts easier

Reply:
Throw an exception saying it's not supported. We do this for some other relations as well: https://github.com/cognitedata/cdp-spark-datasource/blob/master/src/main/scala/cognite/spark/v1/NumericDataPointsRelation.scala#L116